django/core/handlers/asgi.py

```diff
@@ -223,6 +226,17 @@ async def process_request(request, send):
                     pass
         body_file.close()
 
+        if response is not None:
+            # call response.close in sync thread - this invokes the request_finished
+            # signal and ensures expected cleanup code is called
+            await sync_to_async(response.close, thread_sensitive=True)()
+        else:
+            # response is None - process_request was cancelled during or before
+            # run_get_response, so we cannot call response.close
+            # However, processing could have acquired resources that need
+            # to be released by handling the request_finished signal
+            await signals.request_finished.asend(sender=self)
```

Comment:
I think we should a class not an instance
```suggestion
            await signals.request_finished.asend(sender=self.__class__)
```